src/state/nftMarket/hooks.ts

```diff
@@ -105,6 +109,35 @@ export const useUserNfts = (): UserNftsState => {
   return useSelector((state: State) => state.nftMarket.data.user)
 }
 
+export const useApprovalNfts = (nftsInWallet: NftToken[]) => {
+  const nftApprovalCalls = useMemo(
+    () =>
+      nftsInWallet.map((nft: NftToken) => {
+        const { tokenId, collectionAddress } = nft
+
+        return {
+          address: collectionAddress,
+          name: 'getApproved',
+          params: [tokenId],
+        }
+      }),
+    [nftsInWallet],
+  )
+
+  const { data } = useSWRMulticall(erc721Abi, nftApprovalCalls)
+
+  const approvedTokenIds = Array.isArray(data)
+    ? data
+        .flat()
+        .reduce(
+          (acc, address, index) => ({ ...acc, [nftsInWallet[index].tokenId]: getPancakeProfileAddress() === address }),
```

Comment:
This should check against user address no?

Reply:
`nftsInWallet` is already owned by user address. This check is for NFTs enabled to be used in Profile contract.

Reply:
This i agree, but contract returns the address that approves the token not the pancake profile address afaik. Correct me if i am wrong

Reply:
<img width="818" alt="Screen Shot 2022-01-20 at 20 32 47" src="https://user-images.githubusercontent.com/97418926/150348513-83f457b0-0787-43af-97a5-5630afe4a3ee.png">

The approve logic is `approvalList[tokenId] = profileAddress`. Then the `getApprove` function would return `profileAddress` by `tokenId` 

You could check the contract for more detail. https://bscscan.com/address/0xDf7952B35f24aCF7fC0487D01c8d5690a60DBa07#code

Reply:
Sorry my bad :) I confused my address with profile address. :)